SPECS/mariner-release/mariner-release.spec

```diff
@@ -52,6 +52,11 @@ cat > %{buildroot}%{_sysconfdir}/issue.net <<- EOF
 Welcome to CBL-Mariner %{mariner_release_version} (%{_arch})
 EOF
 
+mkdir -p %{buildroot}%{_rpmconfigdir}/macros.d
+cat > %{buildroot}%{_rpmconfigdir}/macros.d/macros.dist <<- EOF
+%%dist %{dist}
+EOF
+
```

Comment:
2 ideas:

1. Move it to one of the `macros*` files installed with `mariner-rpm-macros` (I'm thinking either `macros` or some new one). This way we keep macro definitions in one place, so they are still easy to analyze/discover when searching through our repo. Also, if anyone is trying to build on Mariner but not with out tooling, it's a reasonable requirement to have `mariner-rpm-macros` installed.
2. I'm guessing we'd still like to avoid having to update the macro for each release of Mariner, so we could do something along the lines of `%dist @DIST_TAG@` in the file and then `sed` during the build of the package.

What do you think?

Reply:
Ad 1. I think I found 2 core specs where the macros file is created inside a spec (I searched for `\s>?>.*macro` through all spec files). Not great but I think we can/should fix that, so they don't serve as precedence, like you mentioned. There's 3 more such specs in `SPECS-EXTENDED` I believe.

Ad 2. You're right, that `%{dist}` should be resolved during build time but if we use spec sources instead of dynamically building the macro files in the spec, then we need to update them with this resolved value. Take a look at what `multilib-rpm-config.spec` is doing with its `SOURCE1` inside the `%build` section, to see what I had in mind.

Reply:
For 1), I ran `tdnf provides /usr/lib/rpm/macros.d/*` to check what packages are claiming to provide macro files.

Bit of cleanup gives:
```
cmake
color-filesystem
cups-devel
fdupes
firewalld-filesystem
fontpackages-devel
httpd-devel
javapackages-filesystem
javapackages-local-bootstrap
javapackages-tools
kf5-rpm-macros
lua-rpm-macros
lua-srpm-macros
mariner-check-macros
mariner-rpm-macros
meson
multilib-rpm-config
ninja-build
perl-macros
pkgconf
pyproject-rpm-macros
qt5-qtbase-common
qt5-rpm-macros
ruby
rubygems-devel
systemd-bootstrap-rpm-macros
systemd-rpm-macros
```

For 2, sounds good. I'll shift over to that flow.